Decoupled docs (#7880)
pantheon-systems/documentation: source/content/guides/decoupled/overview/04-site.md

---
title: Pantheon Front-End Sites
subtitle: Site Creation
description: Create and configure a Front-End Site.
tags: [webops, workflow, decoupled]
contributors: [joa-pan, backlineint, cobypear, hckia]
layout: guide
showtoc: true
permalink: docs/guides/decoupled/overview/site
anchorid: site
reviewed: "2023-03-16"
contenttype: [guide]
innav: [false]
categories: [create]
cms: [decoupled]
audience: [development]
product: [decoupled]
integration: [--]
---

This section provides configuration information for Front-End Sites.

## Prerequisites

* The **Decoupled** tab is enabled in your dashboard after onboarding.

* Your site repository is in GitHub.

> * You have a domain to use for the Front-End site.

Reviewer: This should not be necessary to create a Front-End Site

> * You will use Next.js or Gatsby as the frontend framework.

Reviewer: Is there a way we can signal that any JavaScript based runtime, or any static site generator that outputs html/css/js is useable, but we only fully support Next.js and Gatsby?

Reviewer 2: Added!

* (Optional). You are using the latest version of Drupal, or WordPress for the CMS.

* (Optional). If using a pre-configured starter kit template, you will use one of the following combinations:

    * [Drupal and Next.js](/guides/decoupled/drupal-nextjs-frontend-starters/)

    * [WordPress and Next.js](/guides/decoupled/wp-nextjs-frontend-starters/)

    * [WordPress and Gatsby](/guides/decoupled/wp-gatsby-frontend-starters/)

>     > **NOTE**: Other starter kit configurations are not currently supported.

Reviewer: Same here. Is it possible to show that other configs are _possible_, however they may not be fully _supported_?

### Compatibility

You can create an application from scratch by importing a repository or you can use the Pantheon starter kit templates to develop a Front-End Site.

> You can also create a site without a CMS. This is done by importing the repository and using Next.js or Gatsby as the static site generator.

Reviewer: This is really just a nitpick so feel free to disregard :)

```suggestion
You can also create a site without a CMS. This is done by utilizing the import repository feature during site creation.
```

I don't love the faux constraint of 'Next.js or Gatsby'. I know there is some future work to be done to make the drop down menu more accurate as to the type of rendering, not the framework (at the moment next = ssr, gatsby = ssg). I'm wondering if changing this message here a bit can clear up some ambiguity? Or will this cause more confusion?

Reviewer 2: Committed!

Supported Versions:

|  Tooling   |      Version    |
|  :---:     |       :---:     |
|  Gatsby    |    4 or higher  |
| Next.js    |   12 or higher  |
|  Drupal    |   9 or higher   |
| WordPress  |   5.9 or higher |

## Create a Front-End Site

Select the backend or frontend guide below that meets your configuration requirements and follow the instructions carefully.

### Backend Starters

- [Drupal Backend Starter](/guides/decoupled/drupal-backend-starters)
- [WordPress Backend Starter](/guides/decoupled/wp-backend-starters)

### Frontend Starters

- [Drupal + Next.js Frontend Starter](/guides/decoupled/drupal-nextjs-frontend-starters)
- [WordPress + Next.js Frontend Starter](/guides/decoupled/wp-nextjs-frontend-starters)
- [WordPress + Gatsby Frontend Starter](/guides/decoupled/wp-gatsby-frontend-starters)

### No Starter

> - [Create a Front-End Site without a Starter](/guides/decoupled/no-starter-kit)

Reviewer: I could see this section as its own page, and instead of linking to the guides, putting the guides in a tabbed section.

    * [Drupal and Next.js](/guides/decoupled/drupal-nextjs-frontend-starters/)

    * [WordPress and Next.js](/guides/decoupled/wp-nextjs-frontend-starters/)

    * [WordPress and Gatsby](/guides/decoupled/wp-gatsby-frontend-starters/)

    <Alert title="Note"  type="info" >

    Other configurations are possible but not fully supported.

    </Alert>

### Compatibility

You can create an application from scratch by importing a repository or you can use the Pantheon starter kit templates to develop a Front-End Site.

You can also create a site without a CMS. This is done by utilizing the import repository feature during site creation.

Supported Versions:

|  Tooling   |      Version    |
|  :---:     |       :---:     |
|  Gatsby    |    4 or higher  |
| Next.js    |   12 or higher  |
|  Drupal    |   9 or higher   |
| WordPress  |   5.9 or higher |

## Before You Use Pantheon Starter Kits

The following components are required to create and configure a new decoupled project using a Front-End Site starter kit:

1. Install [Composer](https://getcomposer.org/download/) globally.
    * Composer is required for the CMS backends.

> 1. Install [Terminus](/terminus/install) version 3.0.0 or higher.

Reviewer 3: @whitneymeredith using Terminus and also the following three items about tokens are optional. This is presented as required.

Reviewer 2: Updated! Thanks!!
